services/cd-service/pkg/service/git.go

```diff
@@ -256,8 +256,14 @@ func (s *GitServer) ReadSyncStatuses(ctx context.Context) (*api.GetGitSyncStatus
 			return err
 		}
 		response.AppStatuses = toApiStatuses(append(unsyncedStatuses, syncFailedStatuses...))
+
+		err = repository.MeasureGitSyncStatus(len(unsyncedStatuses), len(syncFailedStatuses))
```

Comment:
Why are we sending these metrics both in the `ReadSyncStatus` method and in `processEsls`? isn't the `processEsls` enough?
I think we should update the metrics only when the values are changing, not when we're reading them.